### ISSUE TITLE
Can I inject a script before slave run container?

### ISSUE THREAD

**Issue by mingqi** [mesosphere/marathon#1001]
I post this question in Google group, also post here.

Our docker container is depend on AWS EBS. We want to create a EBS volume and mount on slave when launch a task run a container. That require to customize the Marathon's executor, let Marathon's executor to run a script which prepare EBS volume before Docker container run. is possible to do that?  I look through document and find out the 'executor' parameter in API seem can do that. What's mean the 'executor' parameter and how to fill it? thanks

**Comment by aquamatthias** [comment]
Hey @tnachen can you say something to this?

**Comment by tnachen** [comment]
Hi @mingqi, you like to create a EBS volume on every new task?
Currently there isn't a way to pre-run some command before starting a new task, there is an issue open and I believe someone is working on adding a hook so that you can perform custom action on every docker container task on a per slave basis.

**Comment by AI-PaaS** [comment]
How is this PR going?  If there are preCmd and postCmd in the api, then we can dynamic create volum e for multi instance and other things. Also, after docker stop, we can make some volume clean work.

**Comment by AI-PaaS** [comment]
+1

**Comment by AI-PaaS** [comment]
preCmd and postCmd should be run in the host, not in the docker container.

**Comment by SolarisYan** [comment]
+1

在 2016-02-25 20:54:01，"AI-PaaS" <notifications@github.com> 写道：


+1

**Comment by junneyang** [comment]
+1

**Comment by tnachen** [comment]
We have hooks avaialble in Mesos now that can run commands before every docker container launched, @aquamatthias is there any Marathon support for pre commands now?

**Comment by mingqi** [comment]
My solution is write a docker script and use it in mesos slave --docker option. The script can do something before docker run every time.

**Comment by junneyang** [comment]
@mingqi
could you pease give more instructoons？
thanks

**Comment by aquamatthias** [comment]
@tnachen this is currently not supported in Marathon. Can you point me to the proto or an example?

**Comment by junneyang** [comment]
i suggest the hook solution like this:
{
    "id": "/develop/test/myapp",
**    "preCmd": "some initial commands on the slave node",                                                              1**
    "cmd": "env && cd applications/ && chmod 755 run.sh && ./run.sh",  
**    "postCmd": "some post commands on the slave node",                                                                 2**
    "container": {
                "type": "DOCKER",
                "docker": {
                   ...
**                   "preCmd": "some initial commands before the container's CMD or ENTRYPOINT",         3**
**                   "postCmd": "some post commands after the container is running"                                   4**
                   ...
            },
            "env": {
            },
}

preCmd/postCmd  1 && 2 run on the slave node
preCmd/postCmd  3 && 4 run within the container

**Comment by kopax** [comment]
Is this still evaluated?

**Comment by matthewdfuller** [comment]
Any updates on this? We have a container that we want to run via Marathon but need to execute a script on the host first. The `preCmd` option above would be perfect - but it needs to run on the host, not in the Mesos agent sandbox.

**Comment by rsiebeling** [comment]
Any updates on this? I would like to have the preCmd option as given above. Same as Matthew, it needs to be run on the host...

**Comment by jasongilanfarr** [comment]
We do not support this at this time. You could in theory have a app with
the same constraints that runs and then sleeps forever and have your app
depend on this, but you'd need to update both everytime either one updated
in order to achieve your desired result.

**Comment by meichstedt** [comment]
Note: This issue has been migrated to https://jira.mesosphere.com/browse/MARATHON-3931. For more information see https://groups.google.com/forum/#!topic/marathon-framework/khtvf-ifnp8.
